### PR TITLE
Fix wrong positive with complex structure assignation

### PR DESCRIPTION
**Pull request by smortex**
Just hit a bug in the plugin.  Here is a fix :wink:

### CODE REVIEW

**Comment by smortex** [comment]
Poke @deanwilson

File: spec/puppet-lint/plugins/puppet-lint_duplicate_class_parameters_spec.rb [hunk 1]
```diff
@@ -162,4 +162,22 @@ class duplicated_type(
     end
   end
 
+  context 'class with a structure using a variable and assigned in another variable' do
+    let(:code) do
+      <<-EOS
+        class complex_structure_assignation (
+          Optinal[String] $hostname = undef,
```
Comment (deanwilson) [inline spec/puppet-lint/plugins/puppet-lint_duplicate_class_parameters_spec.rb]: Should this be `Optional` ?

**Comment by deanwilson** [comment]
This is looking good, thank you, especially for the test case. I have one small comment that i've made on the PR.

File: spec/puppet-lint/plugins/puppet-lint_duplicate_class_parameters_spec.rb (hunk 1 above)
Comment (smortex) [reply spec/puppet-lint/plugins/puppet-lint_duplicate_class_parameters_spec.rb]: Yes indeed!  I amended the commit to fix the issue.

**Comment by smortex** [comment]
The issue you spotted is now fixed.  Thanks for the review!

**Comment by deanwilson** [comment]
@smortex this is now updated (#13 and pushed https://rubygems.org/gems/puppet-lint-duplicate_class_parameters-check) thank you for the code and fixing the one issue.

**Pull request merged**